plugins/gen-design-doc/aac_gen_design_doc/aac_gen_design_doc_impl.py

```diff
@@ -0,0 +1,103 @@
+"""AaC Plugin implementation module for the aac-gen-design-doc plugin."""
+
+import os
+import sys
+
+from iteration_utilities import flatten
+from jinja2 import Template
+
+from aac import parser, util
+from aac.template_engine import (
+    TemplateOutputFile,
+    generate_template,
+    load_templates,
+    write_generated_templates_to_file,
+)
+
+plugin_version = "0.0.1"
+default_template_file = "templates/system-design-doc.md.jinja2"
+
+
+def gen_design_doc(architecture_files: str, output_directory: str, template_file: str):
+    """
+    Generate a System Design Document from Architecture-as-Code models.
+
+    Args:
+        `architecture_files` <str>: A comma-separated list of yaml file(s) containing the modeled
+                                      system for which to generate the System Design document.
+        `output_directory` <str>: The directory to which the System Design document will be written.
+        `template_file` <str>: The name of the template file to use for generating the document. (optional)
+    """
+    first_arch_file, *other_arch_files = architecture_files.split(",")
+    parsed_models = _get_parsed_models([first_arch_file] + other_arch_files)
+
+    loaded_templates = load_templates(__package__)
+
+    template_file = template_file if template_file else default_template_file
+    template_file_name = os.path.basename(template_file)
+
+    # TODO: Find a better solution to select between available templates.
+    selected_template, *_ = [t for t in loaded_templates if template_file_name == t.name]
+
+    output_filespec = _get_output_filespec(
+        first_arch_file, _get_output_file_extension(template_file_name)
+    )
+
+    template_properties = _make_template_properties(parsed_models, first_arch_file)
+    generated_document = _generate_system_doc(
+        output_filespec, selected_template, template_properties
+    )
+    write_generated_templates_to_file([generated_document], output_directory)
+
+    print(f"Wrote system design document to {os.path.join(output_directory, output_filespec)}")
+
+
+# TODO: We really need this try/except code in a separate function
```

Comment:
Is this TODO something that could be addressed in this PR, or that could be moved to an issue?

Reply:
Probably a separate PR (as tech debt) because it will be a refactor for the parser/validator. We can discuss this more later, if needed.

Reply:
Can you capture the changes you'd like to see in an issue then remove this TODO?

I'd like to start cutting down the inline TODOs.
```suggestion
```

Reply:
Yeah, I'll do that.

Reply:
I'll comment on this to bring it into this last-round review, but just delete the todo comment when you've made the issue.